### DISCUSSION TITLE
How can I avoid exposing the OpenAI API Key to the public?

### DISCUSSION THREAD

**Discussion by RodrigoBertotti** [anse-app/anse#93] [Q&A]
Hi, I'm new to Anse and Astro

Is there a risk of exposing the OpenAI key by putting it in the on .env file? If so, how can I secure the OpenAI key?

Thank you

**Comment by ddiu8081** [comment]
Yes, you can put the API Key in `.env` file, but never upload it. It should be excluded from git version control.

Anse does not currently have an environment variable feature, but [chatgpt-demo](https://github.com/anse-app/chatgpt-demo) does. Deployment platforms such as Vercel can manually enter environment variables, which you should fill in there.

<img width="1126" alt="CleanShot 2023-08-28 at 22 38 51@2x" src="https://github.com/anse-app/anse/assets/1998168/ab816ccd-2c1e-4d1b-9980-1a5b827e64c8">
